### PR TITLE
Tabbed channels

### PR DESCRIPTION
**Pull request by ZeWaka**
This will need a bit of tweaking to work nice with other peoples recent additions.

BY MINIZIUM OR LOST_PENGUIN

### CODE REVIEW

File: robin.user.js [hunk 1]
```diff
@@ -72,30 +72,44 @@
...
+    function hasChannel(source)
+    {
+        channel_array = getChannelList();
         source = String(source).toLowerCase();
 
-        var idx = channel_array.length;
-        while(idx-- > 0) {
-            var current_chan = String(channel_array[idx]).toLowerCase().trim();
+        for (idx = 0; idx < channel_array.length; idx++)
+        {
```
Comment (daegalus) [inline robin.user.js:107]: This hashChannel code was reverted from when I made the changes. I had it as a much more accurate function in my pull request here: https://github.com/daegalus/robin-grow/blob/master/robin.user.js#L66

Switching back will fix the ordering problem we have in the channel filter, and its more accurate.

Comment (dashed) [inline robin.user.js:107]: yep. using the function used on master.

**Comment by daegalus** [comment]
Bugs found so far on my end:
- trim bug where it trims an extra time, removing the channel name of characters from the beginning of the message
- Because robin-grow only checks general for activity, it force refreshes every so often because it thinks there is no activity.

File: robin.user.js [hunk 2]
```diff
@@ -310,7 +324,9 @@
         var channel_needle = $("#chat-prepend-select").val().trim();
         var source = String($(".text-counter-input").val());
 
-        if(settings.filterChannel && !(source.toLowerCase().startsWith(channel_needle.toLowerCase()))) {
+        if (selectedChannel >= 0 && !(source.toLowerCase().startsWith(channelList[selectedChannel].toLowerCase())))
```
Comment (dashed) [inline robin.user.js:339]: `selectedChannel` is referenced before it is declared and initialized.

File: robin.user.js (hunk 1 above)
Comment (daegalus) [inline robin.user.js:107]: I took care of it and fixed it on master again. this needs a rebase.

File: robin.user.js (hunk 2 above)
Comment (daegalus) [inline robin.user.js:339]: Looking at the code, it actually is. Don't forget that Javascript has variable hoisting. So all variables are treated like they are at the top of the function/scope. So technically it is at minimum -1 based on the lines a few lines down.

**Comment by 5a1t** [comment]
This is included in the most recent release.

**Pull request merged**